Framework: Add WatchIgnorePlugin for node_modules

umurkontaci opened the pull request:
This change cuts down the numbers of watchers to set from over 7000 to 500-600.

We are setting watchers for all files in node_modules by default but since these files don't change often and if they change they usually warrant a full restart of the dev server, they are OK to ignore. And by ignoring that folder, we no longer exhaust the available file handles in the OS.


### Before:

```
$ osqueryi 'select count(*), p.name from process_open_files as po, processes as p where p.pid=po.pid group by p.pid order by count(*) desc limit 1'

+----------+----------------+
| count(*) | name           |
+----------+----------------+
| 7384     | node           |
+----------+----------------+
```

### After:
```
$osqueryi 'select count(*), p.name from process_open_files as po, processes as p where p.pid=po.pid group by p.pid order by count(*) desc limit 1'

+----------+----------------+
| count(*) | name           |
+----------+----------------+
| 562      | node           |
+----------+----------------+
```

aduth commented:
Nice discovery. 👍  I'm surprised this is not the default, or exposed more obviously similar to the loader's `include` and `exclude` properties.

umurkontaci commented:
I'm holding off merging this for a bit. I've rebased the branch and it seems we are not setting up tons of watchers anymore. Maybe we updated the versions of something else fixed it?

umurkontaci commented:
Closing as it has been fixed in the upstream (I think, cannot reproduce)
